models/snake_and_ladder.py

```diff
@@ -0,0 +1,86 @@
+__author__ = "Shakti Prasad Lenka"
+__email__ = "sinulenka001@gmail.com"
+
+from random import randrange
+from .player import Player
+
+class SnakeAndLadder(object):
+
+	def __init__(self,boardSize,nDice,snakes,ladder,players = []):
+		self._boardSize = boardSize
+		self._nDice = nDice
+		self._players = players
+		self._snakes = snakes
+		self._ladder = ladder
+		self._nPlayers = len(players)
+		self._nRounds = 0
+
+	def diceThrow(self):
+		value = 0
+		for throw in range(self._nDice):
+			value += randrange(1,6)
+		return value
+
+	def ouput(self,player,diceValue):
+		print("{} rolled a {} and moved from {} to {}".format(
+														player.name,
+															diceValue,
+												player.initialPosition,
+												player.finalPosition))
+
+	def oneRound(self):
+		winnerIndex = -1
+		winner =None
+		success = False
+		for index,player in enumerate(self._players):
+			damnSnake = True
+			yayLadder = True
+
+			#Lets throw dice on behalf of player.name
+			diceValue = self.diceThrow()
+
+			player.initialPosition = player.finalPosition
+			#only change position if final position is less than _boardSize
+			if player.finalPosition + diceValue <=self._boardSize:
+				player.finalPosition += diceValue
+
+
```

Comment:
use python style guid like pep8

Reply:
@gcnit  @akshita27  thanks a lot for all your inputs. :)

Reply:
@gcnit A hearty thanks for all your efforts Gaurav. It would be great if you could provide us the perfect solution from your side. We could compare and keep our short comings in mind. 

From comments on my PR, I could realise there should have two different classes for snake and ladder. One for player and One for the Game. Also I need to use proper,descriptive variable/function names.

Reply:
There is no perfect solution to these types of questions. There could be multiple solutions each with different design decisions.

Any solution that we provide won't be perfect. There will always be some scope of improvement. We'll definitely share a solution that would be good for clearing the machine coding round.